### PR TITLE
Cover metadata cleanup

### PR DESCRIPTION
**Pull request by toots**
This PR:
* Exposes all metadata decode as `file.metadata.<decoder>`
* Moves `file.mp3` functions accordingly
* Factors our flac cover metadata decoding and adds a corresponding encoder
* Uses metadata cover parsing and data URI encoding to encode cover metadata in a unified fashion in `metadata.json.stringify` (that's the fix for #1872)

Event though this is a lot of changes, I believe that it is safe to merge into `v2.0.0` as it only affects peripherals aspects of the software.

### CODE REVIEW

File: libs/metadata.liq [hunk 1]
```diff
@@ -61,34 +61,30 @@ def metadata.comment(m)
...
 
-# Extract cover from metadata.
+# Extract cover from metadata. This function implements cover extraction
+# for the following formats: coverart (ogg) and apic (flac, mp3). 
 # @category Source / Track Processing
 # @param m Metadata from which the cover should be extracted.
+# @param ~coverart_mime Default mime type for `"coverart"` data.
 # @method mime MIME type for the cover.
-def metadata.cover(m)
+def metadata.cover(~coverart_mime="image/jpeg", m)
```
Comment (smimram) [inline libs/metadata.liq:70]:
```suggestion
def metadata.cover(~mime="image/jpeg", m)
```

Comment (smimram) [reply libs/metadata.liq:70]: I don't thik that there is any ambiguity here...

File: libs/metadata.liq [hunk 2]
```diff
@@ -129,12 +125,36 @@ def metadata.cover.remove(m)
   list.assoc.filter(fun (k, _) -> k != "metadata_block_picture" and k != "apic", m)
 end
 
+# Cleanup metadata for export. This is used to remove liquidsoap's internal
```
Comment (smimram) [inline libs/metadata.liq:128]:
```suggestion
# Cleanup metadata for export. This is used to remove Liquidsoap's internal
```

**Review by smimram — APPROVED** [review]
Looks mostly fine to me, thanks!

File: libs/metadata.liq (hunk 1 above)
Comment (toots) [reply libs/metadata.liq:70]: Hmm okay but let's reword the description then.

File: libs/metadata.liq [hunk 3]
```diff
@@ -61,34 +61,30 @@ def metadata.comment(m)
   m["comment"]
 end
 
-# Extract cover from metadata.
+# Extract cover from metadata. This function implements cover extraction
+# for the following formats: coverart (ogg) and apic (flac, mp3). 
 # @category Source / Track Processing
 # @param m Metadata from which the cover should be extracted.
+# @param ~coverart_mime Default mime type for `"coverart"` data.
```
Comment (toots) [inline libs/metadata.liq:68]:
```suggestion
# @param ~mime Default data mime type when unknown
```

File: libs/metadata.liq [hunk 4]
```diff
@@ -61,34 +61,30 @@ def metadata.comment(m)
...
 # @method mime MIME type for the cover.
-def metadata.cover(m)
+def metadata.cover(~mime="image/jpeg", m)
   fname = metadata.filename(m)
   if list.assoc.mem("coverart", m) or list.assoc.mem("metadata_block_picture", m)then
     # We are likely to have an ogg/vorbis file
     cover = list.assoc(default="", "coverart", m)
     if cover != "" then
-      string.base64.decode(cover).{mime=""}
+      null(string.base64.decode(cover).{mime=coverart_mime})
```
Comment (toots) [inline libs/metadata.liq:76]:
```suggestion
      null(string.base64.decode(cover).{mime=mime})
```

File: libs/metadata.liq [hunk 5]
```diff
@@ -129,12 +125,36 @@ def metadata.cover.remove(m)
...
+end
+
 let metadata.json = ()
 
-# Export metadata as JSON object
+# Export metadata as JSON object. Cover art, if found, is extracted using
+# `metadata.cover` and exported with key `"cover"` and exported using
+# `string.data_uri.encode`.
 # @category String 
+# @param ~coverart_mime Default mime type for `"coverart"` data.
```
Comment (toots) [inline libs/metadata.liq:143]:
```suggestion
# @param ~mime Default data mime type when unknown
```

File: libs/metadata.liq [hunk 6]
```diff
@@ -129,12 +125,36 @@ def metadata.cover.remove(m)
...
+# @param ~base64 Encode cover data as `base64`
 # @argsof json.stringify
-def metadata.json.stringify(%argsof(json.stringify), m) =
+def metadata.json.stringify(~coverart_mime="image/jpeg", ~base64=true, %argsof(json.stringify), m) =
```
Comment (toots) [inline libs/metadata.liq:146]:
```suggestion
def metadata.json.stringify(~mime="image/jpeg", ~base64=true, %argsof(json.stringify), m) =
```

File: libs/metadata.liq [hunk 7]
```diff
@@ -129,12 +125,36 @@ def metadata.cover.remove(m)
...
+  c = metadata.cover(coverart_mime=coverart_mime, m)
```
Comment (toots) [inline libs/metadata.liq:147]:
```suggestion
  c = metadata.cover(mime=mime, m)
```

File: libs/metadata.liq (hunk 4 above)
Comment (toots) [inline libs/metadata.liq]:
```suggestion
      null(string.base64.decode(cover).{mime=mime})
```

**Pull request merged**